torch/_subclasses/fake_tensor.py

```diff
@@ -1393,6 +1405,16 @@ def _cache_key(
             # where it wasn't seen on a previous instance of the same op.
             self.shape_env.settings if self.shape_env else None,
         ]
+
+        if isinstance(func, torch._ops.HigherOrderOperator):
+            # For invoke_subgraph, ignore the subgraph arg. We rely on
+            # identifier as a hash for the subgraph. It is Dynamo responsibility
+            # to use same identifier for identical subgraphs. For non-Dynamo
+            # use cases, we will always generate unique identifiers (even if the
+            # subgraphs are identical).
+            if func is torch._higher_order_ops.invoke_subgraph:
```

Comment:
Sorry I kind of forgot the discussion we had on Thursday. I remember we reasoned out that everything was safe? If that's the case, can we include the reasoning here?